Unhelpful error for Mac devs when you run `yarn test` w/o GNU coreutils

When you run `yarn test` on a Mac without GNU coreutils installed the sharness test fails, giving no indication that a lack of coreutils is actually the cause. The error message is repro'd below.

```yarn run v1.12.3
$ node ./config/test.js
tmpdir for backend output: /var/folders/k9/cpqqm8nn3mv2mg025wzb5s4c0000gn/T/sourcecred-test-65700Q7XuPDTFYFRg

Starting tasks
  GO   ensure-flow-typing
  GO   check-stopships
  GO   check-pretty
  GO   lint
  GO   flow
  GO   unit
  GO   backend
 PASS  check-stopships
 PASS  ensure-flow-typing
 PASS  flow
 PASS  backend
  GO   sharness
 FAIL  sharness
Exit code: 2
Contents of stdout:
    *** prove ***
    dummy.t ...................................................... ok
    test_build_static_site.t ..................................... 
    not ok 6 - should fail with a target under a file
    Dubious, test returned 1 (wstat 256, 0x100)
    Failed 1/28 subtests 
    	(less 17 skipped subtests: 10 okay)
    test_js_tests_have_top_level_describe_block_with_filename.t .. ok
    test_load_example_github.t ................................... ok
    
    Test Summary Report
    -------------------
    test_build_static_site.t                                   (Wstat: 256 Tests: 28 Failed: 1)
      Failed test:  6
      Non-zero exit status: 1
    Files=4, Tests=108,  3 wallclock secs ( 0.08 usr  0.02 sys +  0.80 cusr  0.85 csys =  1.75 CPU)
    Result: FAIL
Contents of stderr:
    make: *** [prove] Error 1

 PASS  check-pretty
 PASS  unit
 PASS  lint

Full results
 PASS  ensure-flow-typing
 PASS  check-stopships
 PASS  check-pretty
 PASS  lint
 PASS  flow
Contents of stdout:
    No errors!

 PASS  unit
Contents of stderr:
    PASS src/webutil/createRelativeHistory.test.js
    PASS src/explorer/App.test.js
    PASS src/plugins/git/example/exampleRepo.test.js
    PASS src/explorer/weights/PluginWeightConfig.test.js
    PASS src/plugins/github/graphView.test.js
    fatal: bad object 0000000000000000000000000000000000000000
    PASS src/explorer/weights/WeightConfig.test.js
    fatal: Needed a single revision
    PASS src/graphql/mirror.test.js (8.719s)
    PASS src/plugins/git/loadRepository.test.js (8.853s)
    PASS src/plugins/github/createGraph.test.js
    PASS src/plugins/github/render.test.js
    PASS src/webutil/withAssets.test.js
    PASS src/plugins/github/graphqlTypes.test.js
    PASS src/graphql/generateFlowTypes.test.js
    PASS src/explorer/pagerankTable/Aggregation.test.js
    PASS src/plugins/github/relationalView.test.js
    PASS src/core/graph.test.js
    PASS src/cli/main.test.js
    PASS src/explorer/pagerankTable/Connection.test.js
    PASS src/cli/help.test.js
    PASS src/plugins/git/analysisAdapter.test.js
    PASS src/webutil/Link.test.js
    PASS src/explorer/pagerankTable/Table.test.js
    PASS src/plugins/github/analysisAdapter.test.js
    PASS src/plugins/git/render.test.js
    PASS src/explorer/pagerankTable/TableRow.test.js
    PASS src/explorer/weights/NodeTypeConfig.test.js
    PASS src/explorer/pagerankTable/Node.test.js
    PASS src/explorer/weights/WeightSlider.test.js
    PASS src/plugins/github/nodes.test.js
    PASS src/plugins/github/parseReferences.test.js
    PASS src/explorer/weights/EdgeTypeConfig.test.js
    PASS src/plugins/github/heuristics/mentionsAuthorReference.test.js
    PASS src/plugins/github/parseMarkdown.test.js
    PASS src/cli/load.test.js
    PASS src/explorer/pagerankTable/aggregate.test.js
    PASS src/webutil/checkedLocalStore.test.js
    PASS src/core/address.test.js
    PASS src/plugins/git/graphView.test.js
    PASS src/explorer/state.test.js
    PASS src/webutil/assets.test.js
    PASS src/util/map.test.js
    PASS src/core/version.test.js
    PASS src/explorer/adapters/adapterSet.test.js
    PASS src/cli/sourcecred.test.js
    PASS src/graphql/queries.test.js
    PASS src/cli/testUtil.test.js
    PASS src/graphql/schema.test.js
    PASS src/plugins/git/edges.test.js
    PASS src/plugins/git/nodes.test.js
    PASS src/analysis/nodeScore.test.js
    PASS src/util/null.test.js
    PASS src/homepage/routeData.test.js
    PASS src/plugins/git/createGraph.test.js
    PASS src/plugins/github/edges.test.js
    PASS src/plugins/github/urlIdParse.test.js
    PASS src/core/attribution/markovChain.test.js
    PASS src/explorer/weights/weights.test.js
    PASS src/analysis/weights.test.js
    PASS src/util/pathNormalize.test.js
    PASS src/analysis/weightsToEdgeEvaluator.test.js
    PASS src/analysis/pagerankNodeDecomposition.test.js
    PASS src/cli/analyze.test.js
    PASS src/cli/command.test.js
    PASS src/core/attribution/graphToMarkovChain.test.js
    PASS src/core/repoIdRegistry.test.js
    PASS src/plugins/github/schema.test.js
    PASS src/plugins/git/mergeRepository.test.js
    PASS src/cli/common.test.js
    PASS src/util/compat.test.js
    PASS src/core/trie.test.js
    PASS src/util/dedent.test.js
    PASS src/webutil/memoryLocalStore.test.js
    PASS src/core/repoId.test.js
    PASS src/plugins/github/githubGitGateway.test.js
    
    Test Suites: 74 passed, 74 total
    Tests:       1389 passed, 1389 total
    Snapshots:   138 passed, 138 total
    Time:        21.199s, estimated 22s
    Ran all test suites.

 PASS  backend
Contents of stdout:
    Hash: df7c8003993036f43049
    Version: webpack 3.8.1
    Time: 12699ms
                                Asset     Size  Chunks             Chunk Names
                        sourcecred.js   245 kB       0  [emitted]  sourcecred
           fetchAndPrintGithubRepo.js   121 kB       1  [emitted]  fetchAndPrintGithubRepo
                 createExampleRepo.js  14.6 kB       2  [emitted]  createExampleRepo
    generateGithubGraphqlFlowTypes.js  21.5 kB       3  [emitted]  generateGithubGraphqlFlowTypes
       [1] ./src/core/repoId.js 1 kB {0} {1} [built]
       [4] ./src/util/null.js 2.74 kB {0} {1} {2} [built]
       [9] ./src/graphql/queries.js 5.72 kB {0} {1} [built]
      [10] ./src/plugins/github/schema.js 3.73 kB {0} {1} {3} [built]
      [15] ./src/plugins/git/gitUtils.js 1.97 kB {0} {2} [built]
      [18] ./src/plugins/github/fetchGithubRepo.js 6.25 kB {0} {1} [built]
      [29] ./src/cli/main.js 714 bytes {0} [built]
      [30] ./src/cli/command.js 343 bytes {0} [built]
      [31] ./src/cli/sourcecred.js 1.02 kB {0} [built]
      [58] ./src/tools/entry.js 522 bytes {0} [built]
      [60] ./src/plugins/github/bin/generateGraphqlFlowTypes.js 309 bytes {3} [built]
      [61] ./src/plugins/github/generateGraphqlFlowTypes.js 712 bytes {3} [built]
      [63] ./src/plugins/github/bin/fetchAndPrintGithubRepo.js 1.71 kB {1} [built]
      [64] ./src/plugins/git/bin/createExampleRepo.js 1.99 kB {2} [built]
      [65] ./src/plugins/git/example/exampleRepo.js 4.28 kB {2} [built]
        + 51 hidden modules
Contents of stderr:
    Removing contents of build directory: /var/folders/k9/cpqqm8nn3mv2mg025wzb5s4c0000gn/T/sourcecred-test-65700Q7XuPDTFYFRg

 FAIL  sharness
Exit code: 2
Contents of stdout:
    *** prove ***
    dummy.t ...................................................... ok
    test_build_static_site.t ..................................... 
    not ok 6 - should fail with a target under a file
    Dubious, test returned 1 (wstat 256, 0x100)
    Failed 1/28 subtests 
    	(less 17 skipped subtests: 10 okay)
    test_js_tests_have_top_level_describe_block_with_filename.t .. ok
    test_load_example_github.t ................................... ok
    
    Test Summary Report
    -------------------
    test_build_static_site.t                                   (Wstat: 256 Tests: 28 Failed: 1)
      Failed test:  6
      Non-zero exit status: 1
    Files=4, Tests=108,  3 wallclock secs ( 0.08 usr  0.02 sys +  0.80 cusr  0.85 csys =  1.75 CPU)
    Result: FAIL
Contents of stderr:
    make: *** [prove] Error 1


Overview
Failed tasks:
  - sharness
Final result:  FAILURE 
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/config/test.js
+++ b/config/test.js
@@ -81,6 +81,11 @@ function makeTasks(mode /*: "BASIC" | "FULL" */) {
       deps: [],
     },
     {
+      id: "check-gnu-coreutils",
+      cmd: ["./scripts/check-gnu-coreutils.sh"],
+      deps: [],
+    },
+    {
       id: "backend",
       cmd: [
         "npm",
@@ -101,7 +106,7 @@ function makeTasks(mode /*: "BASIC" | "FULL" */) {
         "--silent",
         {BASIC: "sharness", FULL: "sharness-full"}[mode],
       ]),
-      deps: ["backend"],
+      deps: ["backend", "check-gnu-coreutils"],
     },
   ];
   const extraTasks = [

--- a/scripts/check-gnu-coreutils.sh
+++ b/scripts/check-gnu-coreutils.sh
@@ -1,0 +1,10 @@
+#!/bin/sh
+set -eu
+
+if ! 2>/dev/null >/dev/null readlink -f . ; then
+  >&2 echo "Error: Your environment does not provide GNU coreutils"
+  >&2 echo "You're likely developing on macOS."
+  >&2 echo "Please see the following link for a fix:"
+  >&2 echo "https://github.com/sourcecred/sourcecred/issues/698#issuecomment-417202213"
+  exit 1
+fi

